tests/framework/utils.py

```diff
@@ -108,23 +111,37 @@ def download_images():
             image = self.docker_image_dict[key]
             shell_subprocess_run(f'kind load docker-image {image}')
 
-    def __install_crd_and_operator(self):
-        """Install both CRD and KubeRay operator by kuberay-operator chart"""
-        repo, tag = self.docker_image_dict[CONST.OPERATOR_IMAGE_KEY].split(':')
-        if f"{repo}:{tag}" == CONST.KUBERAY_LATEST_RELEASE:
-            logger.info("Install both CRD and KubeRay operator with the latest release.")
-            shell_subprocess_run(
-                "helm repo add kuberay https://ray-project.github.io/kuberay-helm/"
-            )
-            shell_subprocess_run(
-                f"helm install kuberay-operator kuberay/kuberay-operator --version {tag[1:]}"
-            )
-        else:
-            logger.info("Install both nightly CRD and KubeRay operator by kuberay-operator chart")
-            shell_subprocess_run(
-                f"helm install kuberay-operator {CONST.HELM_CHART_ROOT}/kuberay-operator/ "
-                f"--set image.repository={repo},image.tag={tag}"
-            )
+    def __install_crd_and_operator(self, namespace, patch):
+        """
+        Install both CRD and KubeRay operator by kuberay-operator chart.
+        KubeRay operator will install in the {namespace} with custom config describe in {patch}.
+        The default KubeRay operator config is in kuberay/helm-chart/kuberay-operator/values.yaml.
+        """
+        base_yaml = CONST.HELM_CHART_ROOT.joinpath("kuberay-operator/values.yaml")
+        with open(base_yaml,encoding = "utf-8") as base_fd:
+            with tempfile.NamedTemporaryFile('w') as values_fd:
+                helm_chart_values = yaml.safe_load(base_fd)
+                yaml.safe_dump(patch.apply(helm_chart_values),values_fd)
+                values_yaml = values_fd.name
+                repo, tag = self.docker_image_dict[CONST.OPERATOR_IMAGE_KEY].split(':')
+                if f"{repo}:{tag}" == CONST.KUBERAY_LATEST_RELEASE:
```

Comment:
This branch plans to install KubeRay's latest release, but the logic here seems not.